### DISCUSSION TITLE
Is getServerSideProps being called when we call a page from client side?

### DISCUSSION THREAD

**Discussion by pooryak** [vercel/next.js#44107] [Help] [answered]
I used to have getInitialProps on of my pages and I changed it to getServerSideProps. Since then, sometimes when I come to that page from another one, I see the old data but if I refresh the page or come to that page from the URL it shows the new data.
So I wonder if getServerSideProps just is called when you load it from the server side?

**Comment by icyJoseph** [comment, marked as answer]
On SPA transitions, pages that implement GetServerSideProps, receive their props by:

- Making a call to a next/data endpoint which runs GSSP server side
- The result of the call is `.json()` parsed and passed as props to the page component

What you are describing seems to be caused by a Service Worker. Do you have one? It could also be a bug or a caching issue at some level of your architecture, but let's start by discarding the Service Worker.
